src/label/labelLayoutHelper.ts

```diff
@@ -362,4 +372,318 @@ export function hideOverlap(labelList: LabelLayoutInfo[]) {
             displayedLabels.push(labelItem);
         }
     }
+}
+
+/**
+ * remove label overlaps
+ */
+export function removeOverlap(
```

Comment:
This method can be moved out to an individual file and named with a specific algorithm.